ledger/stateproofverificationtracker.go

```diff
@@ -0,0 +1,340 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package ledger
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/algorand/go-algorand/ledger/store"
+
+	"github.com/algorand/go-deadlock"
+
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+	"github.com/algorand/go-algorand/logging"
+	"github.com/algorand/go-algorand/protocol"
+)
+
+var (
+	errStateProofVerificationContextNotFound = errors.New("requested state proof verification context not found")
+)
+
+type verificationDeleteContext struct {
+	confirmedRound      basics.Round
+	stateProofNextRound basics.Round
+}
+
+type verificationCommitContext struct {
+	confirmedRound      basics.Round
+	verificationContext ledgercore.StateProofVerificationContext
+}
+
+// stateProofVerificationTracker is in charge of tracking context required to verify state proofs until such a time
+// as the context is no longer needed.
+type stateProofVerificationTracker struct {
+	// dbQueries are the pre-generated queries used to query the database, if needed,
+	// to lookup state proof verification context.
+	dbQueries *store.StateProofVerificationDbQueries
+
+	// trackedCommitContext represents the part of the tracked verification context currently in memory. Each element in this
+	// array contains both the context required to verify a single state proof and context to decide whether it's possible to
+	// commit the verification context to the database.
+	trackedCommitContext []verificationCommitContext
```

Comment:
nit: call it delta like other trackers?

Reply:
I'm not sure the word _delta_ represent very well what's stored in this slice and its usage.

How about this instead:
```suggestion
		pendingCommit []verificationCommitContext
		pendingDelete []verificationDeleteContext
```